onnxruntime/core/providers/webnn/builders/model_builder.cc

```diff
@@ -90,12 +99,11 @@ Status ModelBuilder::RegisterInitializers() {
     const auto& name = tensor.name();
     const auto& shape = tensor.dims();
 
-    // Ignore the following tensors:
-    // 1. Empty tensors: optional tensors can be indicated by an empty name.
-    // 2. Tensors in skipped_initializers_: These are tensors that are not used as WebNN Constants.
-    //    Note: Scalar tensors are excluded because ONNX Runtime will optimize same scalar initializers into one.
-    if (name.empty() || (Contains(skipped_initializers_, name) && !shape.empty()))
```

Comment:
Does this `!shape.empty())` line not matter anymore? Is it checked elsewhere?

Reply:
We will count usage for all initializers including the scalar (i.e. `shape.empty()`), `!shape.empty())` is not necessary.